components/google_sheets/actions/add-single-row/add-single-row.mjs

```diff
@@ -69,9 +69,14 @@ export default {
   async run({ $ }) {
     let cells;
     if (this.hasHeaders === "Yes") {
-      cells = Object.keys(this).filter((prop) => prop.startsWith("col_"))
-        .sort()
-        .map((prop) => this[prop]);
+      // TODO: If we could create a variable using this.allColumns in additionalProps, we dont need
+      // to call getSpreadsheetValues here again.
+      const allColumns = [];
+      const { values } = await this.googleSheets.getSpreadsheetValues(this.sheetId.value, `${this.sheetName}!1:1`);
+      for (let i = 0; i < values[0]?.length; i++) {
+        allColumns.push(`col_${i.toString().padStart(4, "0")}`);
+      }
+      cells = allColumns.sort().map((column) => this[column] || "");
```

Comment:
**Optional**
You can reduce here. In that way you can avoid the use of `allColumns` array 

```suggestion
      const { values: rows } = await this.googleSheets.getSpreadsheetValues(this.sheetId.value, `${this.sheetName}!1:1`);
      const [
        headers,
      ] = rows;
      cells = headers.reduce((cells, header, idx) => {
        const propName = header || `col_${idx.toString().padStart(4, "0")}`;
        return cells.concat(this[propName]);
      }, []);
```

**Request**
Also I'm not sure why are you `sort`ing here the array? I'm wondering because I tested it without it and it worked as expected.

Reply:
Humm. I changed the implementation to use two `map`.
Yeah, the `sort` doesn't make sense indeed. I guess I forgot to remove it. Tks!